opensrp-core/src/main/java/org/smartregister/account/AccountHelper.java

```diff
@@ -157,8 +117,61 @@ public static String getCachedOAuthToken(String accountName, String accountType,
     public static AccountManagerFuture<Bundle> reAuthenticateUserAfterSessionExpired(String accountName, String accountType, String authTokenType) {
         Account account = getOauthAccountByNameAndType(accountName, accountType);
         return accountManager.updateCredentials(account, authTokenType, null, null, null, null);
+    }
+
+    /**
+     * A Helper method to check if the Access Token is valid
+     */
+    public static boolean isAccessTokenValid(String accountName, String accountType) {
+        String createdAt = getAccountManagerValue(INTENT_KEY.ACCOUNT_ACCESS_TOKEN_CREATED_AT, accountName, accountType);
+        String accountExpires = getAccountManagerValue(INTENT_KEY.ACCOUNT_REFRESH_TOKEN_EXPIRES_IN, accountName, accountType);
```

Comment:
I think we should be using `ACCOUNT_ACCESS_TOKEN_EXPIRES_IN` to calculate when the access token expires and not the refresh token expiry

Reply:
Good catch. The naming of the method is a bit misleading - I've renamed it. 

We should actually be checking for the `ACCOUNT_REFRESH_TOKEN_EXPIRES_IN` since we use the function to force the user to do a remote login. We only do this if the _Refresh token_ expires. The _Access token_ is renewed automatically if the _Refresh token_ is still valid.